### PR TITLE
Fixing service paths

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: search-resources-api/src/main/java/no/unit/nva/search/SearchHandler.java [hunk 1]
```diff
@@ -137,6 +138,10 @@ private String extractInstitutionNumberFromRequestedOrganization(URI requestedOr
     }
 
     private String getIndexName(RequestInfo requestInfo) {
-        return attempt(() -> requestInfo.getPathParameter(INDEX)).orElseThrow();
+        return attempt(() -> removeAllSlash(requestInfo.getPath())).orElseThrow();
+    }
+
+    protected static String removeAllSlash(String path) {
+        return path.replaceAll(SLASH, BLANK);
```
Comment (janbj) [inline search-resources-api/src/main/java/no/unit/nva/search/SearchHandler.java:145]: This works for now but more complex paths will require more processing.

File: search-resources-api/src/main/java/no/unit/nva/search/SearchHandler.java [hunk 2]
```diff
@@ -137,6 +138,10 @@ private String extractInstitutionNumberFromRequestedOrganization(URI requestedOr
     }
 
     private String getIndexName(RequestInfo requestInfo) {
-        return attempt(() -> requestInfo.getPathParameter(INDEX)).orElseThrow();
+        return attempt(() -> removeAllSlash(requestInfo.getPath())).orElseThrow();
```
Comment (axthosarouris) [inline search-resources-api/src/main/java/no/unit/nva/search/SearchHandler.java:141]: Why attempt? removeAllSlash does not throw any kind of exception.

File: search-resources-api/src/main/java/no/unit/nva/search/SearchHandler.java (hunk 1 above)
Comment (axthosarouris) [reply search-resources-api/src/main/java/no/unit/nva/search/SearchHandler.java:145]: Suggestion:
```
    private String getIndexName(RequestInfo requestInfo) {
        return UnixPath.of(requestInfo.getPath()).getFilename();
    }
```
getFilename returns the last part of the path.

File: search-resources-api/src/test/java/no/unit/nva/search/SearchHandlerTest.java [hunk 3]
```diff
@@ -129,6 +129,12 @@ void shouldNotSendQueryAndReturnForbiddenWhenUserDoesNotHaveTheAppropriateAccess
         assertThat(searchRequest, is(nullValue()));
     }
 
+    @Test
+    void shouldReturnIndexNameWithoutSlash() {
```
[the remark this answers is not in the document]
Comment (axthosarouris) [reply search-resources-api/src/test/java/no/unit/nva/search/SearchHandlerTest.java:133]: Suggestion:
```
@ParameterizedTest(name= "should send request to index specified in path")
    @ValueSource(strings={"messages","doirequests"})
    void shouldReturnIndexNameWithoutSlash(String indexName) throws IOException {
        handler.handleRequest(queryWithoutQueryParameters(indexName), outputStream, context);
        var searchRequest = restHighLevelClientWrapper.getSearchRequest();
         var indices = Arrays.stream(searchRequest.indices()).collect(Collectors.toList());
        assertThat(indices, contains(indexName));
    }
```

**Review by axthosarouris — CHANGES_REQUESTED** [review]
see comments

File: search-resources-api/src/main/java/no/unit/nva/search/SearchHandler.java (hunk 1 above)
Comment (janbj) [reply search-resources-api/src/main/java/no/unit/nva/search/SearchHandler.java]: Agreed

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**